### PR TITLE
feat(skills): add hook-suggester analyzers and /tune skill

### PR DESCRIPTION
**Pull request by paulnsorensen**
## Summary
- **3 new hook-suggester analyzers**: correction-patterns (mines user corrections for SLM rule candidates), retry-loops (detects error→retry chains), permission-tool-waste (finds tools burning turns on permission failures)
- **`/tune` skill**: automated single-rule tuning loop — runs eval, analyzes FP/FN misclassifications, edits the rule prompt, re-evaluates up to 3 iterations targeting ≥95% precision / ≥80% recall
- **Tests**: 82 lines covering all three new analyzers (empty results, suggestion shape, edge cases)

Extracted from `ralph/rule-tuning-v0.1` branch for focused review.

## Test plan
- [ ] `just test tests/test_analyze.py` — new analyzer tests pass
- [ ] `just check` — lint + typecheck + test suite green

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: skills/tune/SKILL.md [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+
+Run eval for this specific rule using the project's eval wrapper:
+
+```bash
+bash ralphs/rule-tuning-v2/run-eval.sh 2>&1 | grep -E "(Evaluating <rule-name>|=== <rule-name>|Accuracy|Precision|Recall|F1|Confusion|Confidence|Misclass|actual=)"
+```
+
+If the run-eval.sh script doesn't support single-rule filtering, use:
+
+```bash
```
[the remark this answers is not in the document]
Comment (paulnsorensen) [reply skills/tune/SKILL.md]: Fixed — replaced the ralph-specific path with the canonical `uv run python -m vaudeville.eval --rules-dir rules_dev --rule <rule-name>` as the sole eval command. The ralph script path was a branch-local convenience that wouldn't survive the merge.

File: skills/tune/SKILL.md [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+name: vaudeville:tune
+description: >
+  Tune a single vaudeville SLM rule to meet precision/recall targets. Runs eval,
+  analyzes misclassifications (FP/FN), edits the rule prompt, and re-evaluates
+  in a loop. Use when the user says "tune rule", "fix rule accuracy", "improve
+  rule", "tune violation-detector", "why is my rule failing", or invokes /tune
+  with a rule name. Requires the vaudeville daemon to be running for inference.
+model: sonnet
+context: fork
+allowed-tools: Bash(uv:*), Bash(bash:*), Read, Edit, Write, Glob
```
[the remark this answers is not in the document]
Comment (paulnsorensen) [reply skills/tune/SKILL.md]: Fixed — replaced `Bash(bash:*)` with `Bash(ls:*)` and added `Grep` (the project's dedicated search tool) rather than `Bash(grep:*)`. Step 1 in the skill uses `Glob` for file listing anyway, so `Bash(bash:*)` was wider than needed.

File: skills/hook-suggester/scripts/analyze.py [hunk 3]
```diff
@@ -352,6 +352,148 @@ def check_repeated_bash_patterns(days, min_occ):
...
+        FROM perm_failures
+        GROUP BY tool_name
+        HAVING count(*) >= {min_occ}
+        ORDER BY failures DESC
+        LIMIT 10;
+    """)
+    if not rows:
+        return None
+    total = sum(int(r["failures"]) for r in rows)
+    examples = [f"{r['tool_name']} ({r['failures']} permission failures)" for r in rows]
```
[the remark this answers is not in the document]
Comment (paulnsorensen) [reply skills/hook-suggester/scripts/analyze.py]: Valid catch — fixed the docstring to match what the query actually computes. The query counts permission failures (not retries), which the title "Tools wasting turns on permission walls" already describes correctly. The docstring now reads: "Find tools that hit permission errors, burning turns a PreToolUse hook could intercept."

The full retry-detection query you suggested is a valid future enhancement if we want to verify the tool was actually called again after the failure, but that's a richer signal than what this check needs to surface actionable suggestions.

File: skills/hook-suggester/scripts/analyze.py [hunk 4]
```diff
@@ -352,6 +352,148 @@ def check_repeated_bash_patterns(days, min_occ):
...
+        WHERE type = 'user'
+          AND userType = 'external'
+          AND length(message::VARCHAR) < 200
+          AND timestamp::DATE >= CURRENT_DATE - INTERVAL '{days}' DAY
+          AND (message::VARCHAR ILIKE '%no,%'
+               OR message::VARCHAR ILIKE '%wrong%'
+               OR message::VARCHAR ILIKE '%that''s not%'
+               OR message::VARCHAR ILIKE '%stop %doing%'
+               OR message::VARCHAR ILIKE '%I said%'
+               OR message::VARCHAR ILIKE '%not what I%')
```
[the remark this answers is not in the document]
Comment (paulnsorensen) [reply skills/hook-suggester/scripts/analyze.py:372]: Addressed with `AND typeof(message) = 'VARCHAR'` — in DuckDB, plain text messages load as VARCHAR-typed values; structured JSON messages (content arrays, tool results) load as JSON-typed values. The typeof filter excludes the JSON case before any cast or ILIKE runs.

The `json_each(message, '$.content')` approach would be needed if we wanted to extract text *from* structured messages, but this analyzer is specifically looking for short, plaintext user corrections — not multi-part tool-result payloads. Excluding structured messages by type is the correct behavior here, not a gap.
